packages/repl/src/lib/Workspace.svelte.ts

```diff
@@ -425,13 +425,9 @@ export class Workspace {
 			throw new Error('Workspace must have at least one file');
 		}
 
-		if (selected) {
-			const file = files.find((file) => is_file(file) && file.name === selected);
-
-			if (!file) {
-				throw new Error(`Invalid selection ${selected}`);
-			}
-			this.#select(file as File);
+		const matching_file = selected && files.find((file) => is_file(file) && file.name === selected);
+		if (matching_file) {
+			this.#select(matching_file as File);
 		} else {
 			this.#select(first);
 		}
```

Comment:
If we trust the types (`selected` can't be undefined because `this.#current` is never `nullish`), this can be further simplified to:

```suggestion
		const file = files.find((file) => is_file(file) && file.name === selected) as File;
		this.#select(file ?? first);
```

and the `?` in this method's default argument can be removed.

Reply:
Ah yes, you're right, I completely missed the initialization and was only looking at assignments to `#current`.